auditbeat/module/audit/file/metricset_test.go

```diff
@@ -45,3 +49,258 @@ func getConfig(path string) map[string]interface{} {
+		if runtime.GOOS == "windows" {
+			t.Skip("skip chown on windows")
+		}
+
+		gid := changeGID(t, txt2)
+		event := readTimeout(t, events)
+		assertSameFile(t, txt2, event.Path)
+		assert.Equal(t, AttributesModified.String(), event.Action)
+		assert.EqualValues(t, gid, event.Info.GID)
+	})
+
+	t.Run("deleted", func(t *testing.T) {
+		if err = os.Remove(txt2); err != nil {
+			t.Fatal(err)
+		}
+
+		event := readTimeout(t, events)
+		assert.Equal(t, Deleted.String(), event.Action)
+	})
+
+	// Create a sub-directory.
+	subDir := filepath.Join(dir, "subdir")
+	t.Run("dir created", func(t *testing.T) {
+		if err = os.Mkdir(subDir, 0755); err != nil {
+			t.Fatal(err)
+		}
+
+		event := readTimeout(t, events)
+		assertSameFile(t, subDir, event.Path)
+	})
+
+	// Test that it does not monitor recursively.
+	subFile := filepath.Join(subDir, "foo.txt")
+	t.Run("non-recursive", func(t *testing.T) {
+		if err = ioutil.WriteFile(subFile, []byte("foo"), fileMode); err != nil {
+			t.Fatal(err)
+		}
+
+		assertNoEvent(t, events)
+	})
+
+	// Test moving a file into the monitored dir from outside.
+	var moveInOrig string
+	moveIn := filepath.Join(dir, "test3.txt")
+	t.Run("move in", func(t *testing.T) {
+		f, err := ioutil.TempFile("", "test3.txt")
+		if err != nil {
+			t.Fatal(err)
+		}
+		f.WriteString("move-in")
+		f.Sync()
+		f.Close()
+		moveInOrig = f.Name()
+
+		if err = os.Rename(moveInOrig, moveIn); err != nil {
+			t.Fatal(err)
+		}
+
+		event := readTimeout(t, events)
+		assert.Equal(t, Created.String(), event.Action)
+		assertSameFile(t, moveIn, event.Path)
+	})
+
+	// Test moving a file out of the monitored dir.
+	t.Run("move out", func(t *testing.T) {
+		if err = os.Rename(moveIn, moveInOrig); err != nil {
+			t.Fatal(err)
+		}
+		defer os.Remove(moveInOrig)
+
+		event := readTimeout(t, events)
+		assertSameFile(t, moveIn, event.Path)
+		if runtime.GOOS == "windows" {
+			assert.Equal(t, Deleted.String(), event.Action)
+		} else {
+			assert.Equal(t, Moved.String(), event.Action)
+		}
+	})
+}
+
+// readTimeout reads one event from the channel and returns it. If it does
+// not receive an event after one second it will time-out and fail the test.
+func readTimeout(t testing.TB, events <-chan Event) Event {
+	select {
+	case e, ok := <-events:
+		if !ok {
+			t.Fatal("failed reading from event channel")
+		}
+		t.Logf("%+v", buildMapStr(&e).StringToPrint())
+		return e
+	case <-time.After(time.Second):
+		t.Fatalf("%+v", errors.Errorf("timed-out waiting for event"))
+	}
+
+	return Event{}
+}
+
+// assertNoEvent asserts that no event is received on the channel. It waits for
+// 250ms.
+func assertNoEvent(t testing.TB, events <-chan Event) {
+	select {
+	case e := <-events:
+		t.Fatal("received unexpected event", e)
+	case <-time.After(250 * time.Millisecond):
+	}
+}
+
+// assertSameFile asserts that two files are the same.
+func assertSameFile(t testing.TB, f1, f2 string) {
+	if f1 == f2 {
+		return
+	}
+
+	info1, err := os.Stat(f1)
```

Comment:
Should you use LStat here? if one of the params is a symlink to the file, do you expect this check to return true or false?